main.rb

```diff
@@ -0,0 +1,68 @@
+module Enumerable
+  def my_each()
+    for i in self
+      yield(i)
+    end
+  end
+
+  def my_each_with_index()
+    for i in (0...self.length)
+      yield(self[i],i)
+    end
+  end
+
+  def my_select()
+    new_array =[]
+    self.my_each do |item|
+      new_array << item if yield(item) 
+    end
+    new_array
+  end
+
+  def my_all?()
+    new_array = self.my_select { |i| yield(i) }
```

Comment:
There is no need for reaching the end of an array if the value is false, please do not reuse my_select method.